src/firmware/arm_firmware/armSketchRev3.ino

```diff
@@ -0,0 +1,763 @@
+        step_in = red*ppr*(ang / 360.0);
+
+        if(ang == 0) {
+        }
+    
+        else if(ang + angles[axis-1] < max_ang && ang + angles[axis-1]>0) { // if the desired move doesn't put the axis outside of its operating range
+            angles[axis - 1] += ang;
+        }
+
+        else {
+            if(ang>0) { // If the input command is to increase position, and last condition failed, we know that the upper arm range is exceeded, so move arm to max position (angle = max)
+                step_in = (red*ppr/360.0)*(max_ang - angles[axis-1]);
+                angles[axis-1] = max_ang; 
+            }
+
+            else { // otherwise, we know the command wants to try to move the rover past the limit switch, so home the axis (angle = 0)
+                step_in = -(red*ppr/360.0)*angles[axis-1];
+                angles[axis-1] = 0; 
+            }
+        }
+
+        step_in = step_in*axisDir[axis-1];
+
+        if(axis == 1) 
+          a1IncMove();
+
+        else if(axis == 2) 
+          a2IncMove();
+
+        else if(axis == 3) 
+          a3IncMove();
+
+        else if(axis == 4) 
+          a4IncMove();
+
+        else if(axis == 5) 
+          a5IncMove();
+
+        else if(axis == 6) 
+          a6IncMove();   
+    } 
+
+   else if (mode == 'A') { // calculates required steps and direction in absolute mode
+      
+        if(ang<=0) { // negative input angle, home arm
+            step_in = 0;
+            angles[axis-1] = 0;
+        }
+
+        else if(ang>=max_ang) { // angle greater than maximum, move axis to maximum
+            step_in = red*ppr/360.0*max_ang;
+            angles[axis-1] = max_ang;
+        }
+    
+        else { // angle within bounds, calculate the step input from angle and move axis
+            step_in = red*ppr*ang/360.0;
+            angles[axis - 1] = ang; 
+        }
+
+        step_in = step_in*axisDir[axis-1];
+
+        if(axis == 1) 
+          a1AbsMove();
+
+        else if(axis == 2) 
+          a2AbsMove();
+
+        else if(axis == 3) 
+          a3AbsMove();
+
+        else if(axis == 4) 
+          a4AbsMove();
+
+        else if(axis == 5) 
+          a5AbsMove();
+
+        else if(axis == 6) 
+          a6AbsMove();
+   }
+   
+   a1LimTrigger();
+   a2LimTrigger();
+   a3LimTrigger();
+   a4LimTrigger();
+   a5LimTrigger();
+   a6LimTrigger();        
+   sendAngles();
+}
+
+
+
+
+
+
+// function for homing full arm 
+void home_arm() {
+
+    initializeHomingMotion();
+    bool completeFlag = false;
+    bool a1StopFlag = false;
+    bool a2StopFlag = false;
+    bool a3StopFlag = false;
+    bool a4StopFlag = false;
+    bool a5StopFlag = false;
+    bool a6StopFlag = false;
+    bool a1FinishFlag = false;
+    bool a2FinishFlag = false;
+    bool a3FinishFlag = false;
+    bool a4FinishFlag = false;
+    bool a5FinishFlag = false;
+    bool a6FinishFlag = false;
+
+    while (!completeFlag) {
```

Comment:
Be consistent with the indentation in this while loop so that the code is easier to read